internal/resources/templates/layout.go

```diff
@@ -35,7 +35,11 @@ var Layout = `
 								<a class="item" href="{{.GinURL}}">Back to GIN</a>
 								<a class="item" href="/repos">Repositories</a>
 								<a class="item" href="/pubvalidate">One-time validation</a>
-								<a class="item" href="/login">Login</a>
+								{{ if eq .UserName ""}}
+									<a class="item" href="/login">Login</a>
+								{{ else }}
+									<a class="item" href="/logout">{{.UserName}}</a>
```

Comment:
Please change it to "{{.UserName}} (logout)" to make it clear what happens when the link is used.